File: solutions/typescript/secret-handshake/1/secret-handshake.ts
export function commands(input: number): string[] {
  let result = []
  
  input &= 0b00011111;
  if ((input & 0b1) === 0b1) result.push("wink");
  if ((input & 0b10) === 0b10) result.push("double blink");
  if ((input & 0b100) === 0b100) result.push("close your eyes");
  if ((input & 0b1000) === 0b1000) result.push("jump");
  if ((input & 0b10000) === 0b10000) result.reverse();
  
  return result;
}

File: solutions/python/secret-handshake/1/secret_handshake.py
def commands(binary_str):
    if (len(binary_str) != 5):
        raise ValueError("binary_str must be len of 5")
    result = []
    if (binary_str[4] == "1"):
        result.append("wink")
    if (binary_str[3] == "1"):
        result.append("double blink")
    if (binary_str[2] == "1"):
        result.append("close your eyes")
    if (binary_str[1] == "1"):
        result.append("jump")
    if (binary_str[0] == "1"):
        result.reverse()
    return result
